### PR TITLE
Adds support for loading a string config content in KubernetesClientConfiguration

### LINKED ISSUE #82: KubernetesClientConfiguration could support loading from string
My team and I are working in k8s automation solution, where we will administrate several different k8s clusters using k8s API.

For that, we store every config yaml file in our databases in a text field (due to limitations on store it as an object). 
When instantiating `K8SConfiguration`, it would be much better load it directly as a string into the constructor.

### PR DESCRIPTION
**Pull request by allantargino**
Hi guys,

fixes #82 
As related on the mentioned issue, we needed this feature in order to don't create temp files in our server.

Does the changes I've made make sense?

Thanks

### CODE REVIEW

**Comment by tg123** [comment]
LGTM

but I suggest support stream if string is supported 
string, file would just be syntax sugar of stream

**Comment by allantargino** [comment]
Hey @tg123, that's a great suggestion, this way we can support multiple ingress cases.

Great thanks to @ViniciusSouza who authored the changes and commit it.

File: src/Properties/PublishProfiles/FolderProfile.pubxml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (tg123) [inline src/Properties/PublishProfiles/FolderProfile.pubxml]: this file should not be checked in

File: src/KubernetesClient.csproj [hunk 2]
```diff
@@ -11,6 +11,10 @@
 
     <TargetFramework>netstandard1.4</TargetFramework>
     <RootNamespace>k8s</RootNamespace>
+    <Version>1.1.0</Version>
```
Comment (tg123) [inline src/KubernetesClient.csproj]: what those properties for?

File: src/KubernetesClient.csproj [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <PackageVersion>0.2.0-beta</PackageVersion>
+    <PackageVersion>0.2.1-beta</PackageVersion>
```
Comment (tg123) [inline src/KubernetesClient.csproj]: maybe should be managed by @brendandburns

File: src/KubernetesClientConfiguration.ConfigFile.cs [hunk 4]
```diff
@@ -228,5 +280,30 @@ private static K8SConfiguration LoadKubeConfig(FileInfo kubeconfig)
...
+        }
+
+        /// <summary>
+        ///     Loads Kube Config from stream.
+        /// </summary>
+        /// <param name="kubeconfig">Kube config file contents</param>
+        /// <returns>Instance of the <see cref="K8SConfiguration"/> class</returns>
+        private static K8SConfiguration LoadKubeConfig(Stream kubeconfig)
+        {
+            StreamReader sr = new StreamReader(kubeconfig);
```
Comment (tg123) [inline src/KubernetesClientConfiguration.ConfigFile.cs]: using?

File: src/KubernetesClientConfiguration.ConfigFile.cs [hunk 5]
```diff
@@ -50,6 +50,58 @@ public partial class KubernetesClientConfiguration
...
+
+            kubeconfig.Position = 0;
+
+            var k8SConfig = LoadKubeConfig(kubeconfig);
+            var k8SConfiguration = GetKubernetesClientConfiguration(ref currentContext, masterUrl, k8SConfig);
+
+            return k8SConfiguration;
+        }
+
+        private static KubernetesClientConfiguration GetKubernetesClientConfiguration(ref string currentContext, string masterUrl, K8SConfiguration k8SConfig)
```
Comment (tg123) [inline src/KubernetesClientConfiguration.ConfigFile.cs]: why ref?

**Comment by brendandburns** [comment]
I'm happy to merge this one @tg123 's comments are addressed.

File: src/KubernetesClient.csproj (hunk 3 above)
Comment (ViniciusSouza) [reply src/KubernetesClient.csproj]: We've created a NuGet package with those changes and I have let it there by mistake. 
Fixed

File: src/KubernetesClient.csproj (hunk 2 above)
Comment (ViniciusSouza) [reply src/KubernetesClient.csproj]: We've created a NuGet package with those changes and I have let it there by mistake. 
Fixed

File: src/Properties/PublishProfiles/FolderProfile.pubxml (hunk 1 above)
Comment (ViniciusSouza) [reply src/Properties/PublishProfiles/FolderProfile.pubxml]: We've created a NuGet package with those changes and I have let it there by mistake. 
fixed, by removing the folder.

File: src/KubernetesClientConfiguration.ConfigFile.cs (hunk 4 above)
Comment (ViniciusSouza) [reply src/KubernetesClientConfiguration.ConfigFile.cs]: fixed

File: src/KubernetesClientConfiguration.ConfigFile.cs (hunk 5 above)
Comment (ViniciusSouza) [reply src/KubernetesClientConfiguration.ConfigFile.cs]: Visual Studio´s refactoring..  fixed.

**Comment by allantargino** [comment]
@tg123 and @brendanburns, all fixed!

**Comment by brendandburns** [comment]
LGTM, @tg123 any more comments?

**Comment by tg123** [comment]
LGTM. better to do a squash merge

**Comment by allantargino** [comment]
Squashed 👍

**Comment by brendandburns** [comment]
Thanks, merging.

**Comment by brendandburns** [comment]
@tg123 fwiw, github now supports "squash and merge" so there's no need for the author to squash...

**Pull request merged**